src/components/connect.js

```diff
@@ -17,9 +27,127 @@ function getDisplayName(WrappedComponent) {
 // Helps track hot reloading.
 let nextVersion = 0
 
-export default function connect(mapPropsToRequestsToProps, options = {}) {
+function connectFactory(defaults = {}) {
+  function connectImpl(map, options = {}) {
+    /* eslint-disable no-console */
+    if (Object.getOwnPropertyNames(options).length > 0 && console && console.warn) {
+      console.warn('The options argument is deprecated in favor of `connect.defaults()` calls. In a future release, support will be removed.')
+    }
```

Comment:
Good idea, and it would avoid the clunky `console && console.warn` checking as well as the ESLint disables. I'll leave it for after review.